superset/datasets/schemas.py

```diff
@@ -42,7 +42,7 @@ def validate_python_date_format(value: str) -> None:
 class DatasetColumnsPutSchema(Schema):
     id = fields.Integer()
     column_name = fields.String(required=True, validate=Length(1, 255))
-    type = fields.String(validate=Length(1, 32))
+    type = fields.String(allow_none=True)
```

Comment:
Do we need to allow `None` here? What happens when we update a dataset setting the type to `None`?

Reply:
The UI already allows to do that. Take an existing dataset, using the legacy editor, set it to blank and save your change, change is persisted. I decided to mimic what's for description.